### DISCUSSION TITLE
change app name

### DISCUSSION THREAD

**Discussion by alirezaimi** [beeware/briefcase#1332] [Q&A]
Hi
I want to  change my app name in android , change pyproject.toml and add parameters to base class object but there is no log ! app name still stuck on hello world !

I add some parameters to this class object and seems to not change anything at all:
```
def main():
    return HomeAutoAPP(formal_name="imiApp", 
                       app_id="org.beeware.imiAPP",
                       app_name="HomeAutoApp")
if __name__ == "__main__":
    app = main()
    app.main_loop()
```

and change pyproject.toml file like this and seems to not change anything at all either :
![image](https://github.com/beeware/briefcase/assets/1029949/fa4bd687-b304-4a8a-9f5d-237efccd1eee)

**Comment by freakboy3742** [comment]
You're hitting 2 known issues with Briefcase/Toga - one of which is a bug, one of which is a borderline bug/feature.

The bug is beeware/toga#2015. Toga's Android backend doesn't currently apply the application name to the main window at time of construction - it uses the app's formal name that was provided in the app metadata file generated by template (look for `app/src/main/res/values/strings.xml` in the generate Android project).

The bug/feature is #472. Briefcase can't update an existing application once it's been generated. The `create` step is templated; once you've rolled out that template, the generated code "locks in" the values in `pyproject.toml` that were present at that time. 
When you combine the two, you end up with an app whose title isn't set based on the value in code, and whose value is hard-coded when the template is rolled out.

beeware/toga#2015 should be addressed in the near future as part of an audit of the core Window functionality in Toga. Addressing #472 is on our longer term roadmap. 

There are 3 potential fixes for right now:

1. Delete your build directory and regenerate your app. In most cases, this *should* be completely safe. You should consider the contents of your build directory to be completely ephemeral - you don't commit this directory to version control; you should be deleting and rebuilding your app regularly. Unfortunately, this isn't a 100% reliable approach at present, as there are a handful of situations where it is necessary to manually update your Android configuration files or generated code. However, if you're *not* modifying your Android project files manually after generation, it should be completely safe.
2. Manually update `strings.xml` to include the new formal name. This is effectively working around #472; it's a safe change to make without committing because the "new" value is what will be generated the next time you create the project from template.
3. Explicitly set the title of the app in code. `main_window.title = "new title"` *does* work on Android; although this call *shouldn't* be required, it's a viable workaround in the short term.
